dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +133,251 @@ async def get_cluster_status(self, req):
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the CPU profile data.
+
+        Raises:
+            ValueError: If the "task_id" parameter is missing in the request query.
+            ValueError: If the "attempt_number" parameter is missing in the request query.
+            ValueError: If the maximum duration allowed is exceeded.
+            ValueError: If the worker begins working on another task during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is an internal server error during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If the CPU Flame Graph information for the task is not found.
+        """
+
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+        pid, ip, state, _ = await self.get_worker_details_for_task(
+            task_id, attempt_number
+        )
+
+        try:
+            self.validate_work_details(pid, ip, state)
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        duration = int(req.query.get("duration", 5))
+        if duration > 60:
+            raise ValueError(f"The max duration allowed is 60: {duration}.")
+        format = req.query.get("format", "flamegraph")
+
+        # Default not using `--native` for profiling
+        native = req.query.get("native", False) == "1"
+        reporter_stub = self._stubs[ip]
+
+        logger.info(
+            "Sending CPU profiling request to {}:{} for {} with native={}".format(
+                ip, pid, task_id, native
+            )
+        )
+
+        reply = await reporter_stub.CpuProfiling(
+            reporter_pb2.CpuProfilingRequest(
+                pid=pid, duration=duration, format=format, native=native
+            )
+        )
+
+        ## Get the new pid and ip again to check if the task is still running and the worker is still working on the task
+        ## Since there is a task scheduling strategy that a worker may run different tasks at different time
+        (
+            new_pid,
+            new_ip,
+            new_state,
+            _,
+        ) = await self.get_worker_details_for_task(task_id, attempt_number)
+
+        try:
+            self.validate_work_details(new_pid, new_ip, new_state)
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        if not reply.success:
+            return aiohttp.web.HTTPInternalServerError(text=reply.output)
+
+        logger.info("Returning profiling response, size {}".format(len(reply.output)))
+        return aiohttp.web.Response(
+            body=reply.output,
```

Comment:
if we cannot alter the body since  - could we add the warnings to the headers?

Reply:
Good idea, I chose to add the warning info as a <p> and display the HTML as: <p> {WARNING} </p> and <svg>{CPU_PROFILE} </svg>, do you think it is a good approach?

Reply:
I don't have a strong opinion on if we want to pass html from backend to front end. To me, it's better if we separate data from rendering